### PR TITLE
[stories][expotool] create tool to generate story app

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

This PR creates a new expotool command `et generate-stories-app` which builds off of the work done on the `generate-bare-app` command. This will allow us to generate a bare app that will render the stories of whichever packages we specified in the command

Example:
`et generate-stories-app expo-checkbox expo-av --outDir ~/tmp --name checkbox-av-stories`

This example will generate a stories app that will include all stories defined in the `expo-av` and `expo-checkbox` packages

# How

Modified the `generate-bare-app` command to write which packages it symlinks and exported a few functions for shared use

Implemented the `generate-stories-app` command which
- configures the native dependencies required for `expo-stories`
- configured metro to work w/ expo stories
- rewrites the `App.js` file to render the story app exported from `expo-stories`

# Test Plan

Run `et generate-stories-app expo-av --name my-story-app --outDir ~/tmp` 
After the command finishes, navigate to that folder and run `expo run:ios` and ensure the story is rendered and that fast refresh is working

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: tools/src/commands/GenerateBareApp.ts [hunk 1]
```diff
@@ -258,18 +275,6 @@ module.exports = config;
   });
 }
 
-async function applyGradleFlipperFixtures({ projectDir }: { projectDir: string }) {
```
Comment (ajsmth) [inline tools/src/commands/GenerateBareApp.ts:261]: this is no longer needed with our new template

File: tools/src/commands/GenerateStoriesApp.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+async function addNativeDependencies({ projectDir }: { projectDir: string }) {
+  const pkgPath = path.resolve(projectDir, 'package.json');
+  const pkg = await fs.readJSON(pkgPath);
+
+  pkg.dependencies['react-native-safe-area-context'] = '4.2.4';
+  pkg.dependencies['react-native-screens'] = '~3.11.1';
+  pkg.dependencies['react-native-svg'] = '12.3.0';
+  pkg.dependencies['react-native-gesture-handler'] = '2.4.2';
+  pkg.dependencies['expo-stories'] = '1.0.1';
```
Comment (barthap) [inline tools/src/commands/GenerateStoriesApp.ts:34]: Hmm, this looks like another place we have to keep in mind when upgrading dependencies. Maybe we could read these versions from some single source of truth, like `bundledNativeModules.json`?

Comment (tsapeta) [reply tools/src/commands/GenerateStoriesApp.ts:34]: Yeah, we definitely don't want to maintain this, so any other way of getting these versions would be great

Comment (ajsmth) [reply tools/src/commands/GenerateStoriesApp.ts]: thank you for the review!

File: tools/src/commands/GenerateStoriesApp.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    'expo',
+    'bundledNativeModules.json'
+  ));
+
+  pkg.dependencies['react-native-safe-area-context'] =
+    bundledNativeModules['react-native-safe-area-context'];
+  pkg.dependencies['react-native-screens'] = bundledNativeModules['react-native-screens'];
+  pkg.dependencies['react-native-svg'] = bundledNativeModules['react-native-svg'];
+  pkg.dependencies['react-native-gesture-handler'] =
+    bundledNativeModules['react-native-gesture-handler'];
```
Comment (bbarthec) [inline tools/src/commands/GenerateStoriesApp.ts:41]: Whenever I have to write the same thing 3rd time I create a helper mechanism doing that.
```suggestion
  [
    'react-native-safe-area-context',
    'react-native-screens',
    'react-native-svg',
    'react-native-gesture-handler'
  ].forEach(packageName => {
    pkg.dependencies[packageName] = bundledNativeModules[packageName];
  });
```

**Review by bbarthec — APPROVED** [review]
This PR looks fine 👍, but I see inconsistencies in logging: there're several functions that have some starting/ending `console.logs` and number of functions that does quite a lot and log nothing 🤔

**Review by barthap — APPROVED** [review]

File: tools/src/commands/GenerateStoriesApp.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+  await fs.outputJson(path.resolve(projectDir, 'package.json'), pkg, { spaces: 2 });
+
+  return await spawnAsync('yarn', [], { cwd: projectDir, stdio: 'ignore' });
+}
+
+async function createMetroConfig({ projectRoot }: { projectRoot: string }) {
+  console.log('Adding metro.config.js for project');
+
+  const template = `// Learn more https://docs.expo.io/guides/customizing-metro
```
Comment (tsapeta) [inline tools/src/commands/GenerateStoriesApp.ts:54]:
```suggestion
  const template = `// Learn more https://docs.expo.dev/guides/customizing-metro
```

**Review by tsapeta — APPROVED** [review]

**Pull request merged**